gtsfm/loader/loader_base.py

```diff
@@ -319,13 +322,23 @@ def get_absolute_pose_priors(self) -> List[Optional[PosePrior]]:
         N = len(self)
         return [self.get_absolute_pose_prior(i) for i in range(N)]
 
-    def get_all_images(self) -> List[Image]:
+    def create_computation_graph_for_images(self) -> List[Delayed]:
         """Creates the computation graph for image fetches.
 
         Returns:
             List of delayed tasks for images.
         """
-        return [self.get_image(i) for i in range(len(self))]
+        N = len(self)
+        annotation = dask.annotate(workers=self._input_worker) if self._input_worker else dask.annotate()
+        with annotation:
+            delayed_images = [dask.delayed(self.get_image)(i) for i in range(N)]
+        return delayed_images
+
+    def get_all_images_as_futures(self, client: Client) -> List[Future]:
+        return [
+            client.submit(self.get_image, i, workers=[self._input_worker] if self._input_worker else None)
```

Comment:
when using an SSHClient, does this always run on the scheduler?

Reply:
is there an `annotate` equivalent?

Reply:
This is the equivalent from what I found out. If someone could run it on a multi-host setup it would be great